Source/missiles.h

```diff
@@ -132,6 +138,29 @@ struct Missile {
 	{
 		return _misource == -1;
 	}
+
+	[[nodiscard]] Player *SourcePlayer()
+	{
+		if (IsNoneOf(_micaster, TARGET_BOTH, TARGET_MONSTERS) || _misource == -1)

```

Comment:
Found an exception to this rule in OperateShrine* and OBJ_MURKYFTN. But it looks like that can just be changed to `TARGET_MONSTERS` as that is there normal value when cast by the player.

Reply:
I think that I updated all of them now. I have a save game near a cryptic shrine and it seems to function ok with the new code.